pyfarm/agent/entrypoints/main.py

```diff
@@ -621,6 +630,27 @@ def remove_pid_file_on_exit():
 
         logger.info("pid: %s" % pid)
 
+        if not isfile(config["run_control_file"]):
+            directory = dirname(config["run_control_file"])
+            if not isdir(directory):
```

Comment:
Rather than checking with isdir, catch OSError\IOError and check errno against EEXIST.